### PR TITLE
Migrate to JUnit framework

### PR DESCRIPTION
**Pull request by wendigo**
All modules are migrated to JUnit. There is no testng anymore :)

### CODE REVIEW

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java [hunk 1]
```diff
@@ -41,12 +42,12 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
 
-@Test(singleThreaded = true)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (ssheikin) [inline bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:45]: static import

File: pom.xml [hunk 2]
```diff
@@ -350,5 +350,21 @@
...
             </plugins>
         </pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <configuration>
+                    <rules>
+                        <bannedDependencies>
+                            <excludes combine.children="append">
```
Comment (ssheikin) [inline pom.xml:360]: excludes -> includes

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java [hunk 3]
```diff
@@ -41,12 +42,12 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.fail;
 
-@Test(singleThreaded = true)
```
Comment (ssheikin) [inline bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:44]: @Test(singleThreaded = true) ->
@TestInstance(PER_CLASS) @Execution(SAME_THREAD)
https://github.com/junit-team/junit5/issues/3849

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 1 above)
Comment (wendigo) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:45]: I wanted to avoid that :)

File: pom.xml (hunk 2 above)
Comment (wendigo) [reply pom.xml:360]: Nope. Include is allow list
Exclude is ban list.

File: concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java [hunk 4]
```diff
@@ -289,122 +288,135 @@ public void testNoStackOverflow()
         Futures.allAsList(futures).get(1, TimeUnit.MINUTES);
     }
 
-    @DataProvider
-    public static Object[][] concurrency()
+    public static int[] testedConcurrency()
```
Comment (ssheikin) [inline concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java:291]: @ParameterizedTest

File: event/src/test/java/io/airlift/event/client/TestEventSubmissionFailedException.java [hunk 5]
```diff
@@ -18,20 +18,25 @@
...
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class TestEventSubmissionFailedException
 {
-    @Test(expectedExceptions = NullPointerException.class, expectedExceptionsMessageRegExp = "causes is null")
+    @Test
     public void testRejectsNull()
     {
-        throw new EventSubmissionFailedException("service", "type", null);
+        assertThatThrownBy(() -> {
```
Comment (ssheikin) [inline event/src/test/java/io/airlift/event/client/TestEventSubmissionFailedException.java:35]: remove curly brackets

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java [hunk 6]
```diff
@@ -202,7 +208,7 @@ protected void doTestConnectTimeout(boolean proxyTest)
         }
     }
 
-    @Test(expectedExceptions = ConnectException.class)
+    @Test
     public void testConnectionRefused()
             throws Exception
```
Comment (ssheikin) [inline http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:213]: remove?

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java [hunk 7]
```diff
@@ -215,7 +221,8 @@ public void testConnectionRefused()
                 .setUri(new URI(scheme, null, host, port, "/", null, null))
                 .build();
 
-        executeExceptionRequest(config, request);
+        assertThatThrownBy(() -> executeExceptionRequest(config, request))
+                .isInstanceOfAny(ConnectException.class, IOException.class, HttpResponseException.class);
```
Comment (ssheikin) [inline http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:225]: why `IOException.class, HttpResponseException.class` added?

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java [hunk 8]
```diff
@@ -262,7 +271,9 @@ public void testBadPort()
                 .setUri(new URI(scheme, null, host, 70_000, "/", null, null))
                 .build();
 
-        executeExceptionRequest(config, request);
+        assertThatThrownBy(() -> executeExceptionRequest(config, request))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessageMatching(".*port out of range.*");
```
Comment (ssheikin) [inline http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:276]: is .* needed?
If yes - I saw somewhere in this PR that the message lacks them.

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 3 above)
Comment (wendigo) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:44]: Well. It worked so...

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 1 above)
Comment (ssheikin) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:45]: why?

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 3 above)
Comment (ssheikin) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:44]: Yeah, it mostly other way around, for methods which where not annotated `@Test(singleThreaded = true)` you have to ensure that they run concurrently.

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java (hunk 6 above)
Comment (wendigo) [reply http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:213]: @ssheikin I'm aware and will cleanup as a follow up

File: concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java (hunk 4 above)
Comment (wendigo) [reply concurrent/src/test/java/io/airlift/concurrent/TestAsyncSemaphore.java:291]: We don't use it in trino as it's against the test guide. Sorry

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java (hunk 7 above)
Comment (ssheikin) [reply http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:225]: I see, it's from child class.

File: event/src/test/java/io/airlift/event/client/TestEventSubmissionFailedException.java (hunk 5 above)
Comment (wendigo) [reply event/src/test/java/io/airlift/event/client/TestEventSubmissionFailedException.java:35]: cant remove on throw :)

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java (hunk 8 above)
Comment (wendigo) [reply http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:279]: This is a direct rewrite so we can improve that as a follow-up

File: http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java (hunk 6 above)
Comment (wendigo) [reply http-client/src/test/java/io/airlift/http/client/AbstractHttpClientTest.java:217]: Added

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 3 above)
Comment (wendigo) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java:44]: Added

File: bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java (hunk 1 above)
Comment (wendigo) [reply bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java]: Added

**Comment by wendigo** [comment]
@findepi do you want to take a look? I'd like to merge it to avoid back-and-forth with other PRs

**Comment by wendigo** [comment]
Surprisingly every push is green (and we build it 3x per push)

**Review by martint — APPROVED** [review]

**Pull request merged**